.github/workflows/tests.yml

```diff
@@ -80,6 +80,12 @@ jobs:
           sudo dpkg -i libssl1.1_1.1.1-1ubuntu2.1_arm64.deb
           sudo dpkg -i libssl-dev_1.1.1-1ubuntu2.1_arm64.deb
 
+        # pytorch doesn't support 3.13 yet, we have to explicitly install 3.12 to prevent
+        # ffmpeg from pulling in 3.13
+      - name: Install python 3.12 (macOS)
+        if: ${{ startsWith(matrix.os, 'macos') }}
+        run: brew install python@3.12
+
```

Comment:
Why not use the setup-python action above?

Reply:
brew will still try to install its version, and complain about a conflicted file